File: Search_for_a_Range.cpp
/*
 * Search_for_a_Range.cpp
 * Copyright (C) 2014 fuel <fuel@fuel-3330>
 *
 * Distributed under terms of the MIT license.
 */

#include <iostream>
#include <algorithm>
#include <FooBee/out>
using namespace std;
class Solution {
public:
    vector<int> searchRange(int A[], int n, int target) {
        int* L = lower_bound(A,A+n,target);
        int* R = upper_bound(A,A+n,target);
        R--;
        if(L==A+n) return {-1,-1};
        else if(*L==target && *R==target) return {int(L-A),int(R-A)};
        else return {-1,-1};
    }
};
int main() {
    Solution s; 
    int A[] = {1};
    cout<<s.searchRange(A,1,1)<<endl;
    return 0;
}

File: Search_for_a_Range.py
#! /usr/bin/env python
# -*- coding: utf-8 -*-
# vim:fenc=utf-8
#
# Copyright © 2014 fuel <fuel@fuel-3330>
#
# Distributed under terms of the MIT license.
import bisect

class Solution:
    # @param A, a list of integers
    # @param target, an integer to be searched
    # @return a list of length 2, [index1, index2]
    def searchRange(self, A, target):
        L = bisect.bisect_left(A,target)
        R = bisect.bisect_right(A,target)-1
        if L<len(A) and A[L]==target:
            return [L,R]
        else:
            return [-1,-1]
